File: C/Martinha.cpp
#include <iostream>
#include <climits>
#include <cmath>
#include <stdio.h>
#define E 2.7182818284590452
//Riemann sums in summation notation (f(2n-1)*largura)
using namespace std;
void integral3X2 (double a, double b, int c){
  double largura = (b-a)/c, altura = (largura/2) + a, resultado = 0;
  for(int i = 0; i < c; i++){
    resultado += 3*(pow(altura,2))*(largura);
    altura += largura;
  }
  printf("\nA area estimada: %f", resultado);
  resultado = 3*(pow(b,3)/3) - 3*(pow(a,3)/3);
  printf("\nA area real: %f", resultado);
}
void integralEX (double a, double b, int c){
  double largura = (b-a)/c, altura = (largura/2) + a, resultado = 0;
  for(int i = 0; i < c; i++){
    resultado += pow(E,altura)*(largura);
    altura += largura;
  }
  printf("\nA area estimada: %.7f", resultado);
  resultado = pow(E,b) - pow(E,a);
  printf("\nA area real: %.7f", resultado);
}
void integralCosX (double a, double b, int c){
  double largura = (b-a)/c, altura = (largura/2) + a, resultado = 0;
  for(int i = 0; i < c; i++){
    resultado += (-1*cos(altura))*(largura);
    altura += largura;
  }
  printf("\nA area estimada: %.7f", resultado);
  resultado = (-1*cos(b)) - (-1*cos(a));
  printf("\nA area real: %.7f", resultado);
}
int main(){
  inicio:
  system("cls");
  double a, b;
  int n, opc = 0;
  try{
    while(opc !=4){
      cout << "Grupos de Funcao - 3\n1. 3X^2\n2. e^x\n3. -cos(x)\n4. Sair\n====> ";
      cin >> opc;
      if(!cin.good())
        throw 1;
      cout << "Digite o limite inicial: ";
      cin >> a;
      cout << "Digite o limite final: ";
      cin >> b;
      cout << "Digite a quantidade de retangulos: ";
      cin >> n;
      switch (opc) {
        case 1:
          integral3X2(a, b, n);
          break;
        case 2:
          integralEX(a, b, n);
          break;
        case 3:
          integralCosX(a, b, n);
          break;
        }
        cout << "\nDigite 1 para reiniciar e 4 para sair: ";
        cin >> opc;
      }
    } catch (...) {
        cout << "Digite um caracter valido\n";
        system("pause");
        cin.clear();
        cin.ignore();
    }
    goto inicio;
  return 0;
}

File: Java/martinha.java
import javax.swing.JOptionPane;
class martinha{
  public static void integral3X2 (float a, float b, int c)
  {
    float largura = ((b-a)/c);
    float altura = ((largura/2) + a);
    float resultado = 0;
    for(int i = 0; i < c; i++) {
      resultado += 3.0f*(Math.pow(altura,2.0f))*(largura);
      altura += largura;
    }
    JOptionPane.showMessageDialog(null, "\nA area estimada: " + resultado + "\nA area real: " + ((float) (3*(Math.pow(b,3)/3) - 3*(Math.pow(a,3)/3))));
    //resultado = (float) (3*(Math.pow(b,3)/3) - 3*(Math.pow(a,3)/3));
    //JOptionPane.showMessageDialog(null, "\nA area real: " + resultado);
  }
  public static void main (String[] args){
    float a, b;
    int n, opc = 0;
    String aux = "";
    while(opc != 4) {
      aux = JOptionPane.showInputDialog("Grupos de Funcao - 3\n1. 3X^2\n2. e^x\n3. -cos(x)\n4. Sair");
      opc = Integer.parseInt(aux);
      aux = JOptionPane.showInputDialog("Digite o limite inicial");
      a = Float.parseFloat(aux);
      aux = JOptionPane.showInputDialog("Digite o limite final");
      b = Float.parseFloat(aux);
      aux = JOptionPane.showInputDialog("Digite a quantidade de retangulos");
      n = Integer.parseInt(aux);
      switch (opc) {
        case 1:
          integral3X2(a, b, n);
          break;
        case 2:
          //integralEX(a, b, n);
          break;
        case 3:
          //integralCosX(a, b, n);
          break;
        default:
          JOptionPane.showMessageDialog(null, "Codigo invalido");
      }
      aux = JOptionPane.showInputDialog("Digite 1 para reiniciar e 4 para sair");
      opc = Integer.parseInt(aux);
    }
  }
}
